timelock-agent/src/main/java/com/palantir/timelock/config/PaxosTsBoundPersisterConfiguration.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.config;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@JsonDeserialize(as = ImmutablePaxosTsBoundPersisterConfiguration.class)
+@JsonSerialize(as = ImmutablePaxosTsBoundPersisterConfiguration.class)
+@Value.Immutable
+public abstract class PaxosTsBoundPersisterConfiguration implements TsBoundPersisterConfiguration {
+
+    public abstract PaxosInstallConfiguration paxos();
```

Comment:
Any thoughts about not allowing this to be configured?

The main benefit of doing this seems to be that you can configure different parameters for the two instances of Paxos:

(A) deciding on cluster leadership,
(B) consensus on the timestamp bound (though we haven't implemented that in PaxosAgent yet).

I'm happy either way, but this is a design decision to consider. If we decide to keep this, then the code in PaxosAgent should respect the config here.

Reply:
Yea I think it makes sense to just have the PaxosTsBoundPersisterConfiguration exist only for the purpose of saying "hey, use paxos to store the timestamp" and then the actual paxos configuration used is the one already mandatorily specified in TimelockInstallConfig

Reply:
Yes, removing the extra paxos block here.